### ISSUE TITLE
Contributing: A bazillion trivial pull-requests, or one ginormous one?

### ISSUE THREAD

**Issue by stevemolloy** [pybee/duvet#14]
I'm very interested in contributing to Duvet, but I'm very new at this and am wondering how the devs would prefer me to work.  Obviously the two extremes I used in the title of this issue are wrong, but what would be best?

Given that Duvet isn't a very active repo, my guess is that the lead dev would rather read a few large(ish) pull requests than swim through loads of small proposals with semi-trivial changes.  Is that true?

What would you recommend?

**Comment by freakboy3742** [comment]
First off - some background: the reason Duvet isn't especially active at the moment is because of [Toga](https://github.com/pybee/toga). I started to hit some limitations in Tk, and the best option I found was to develop a replacement. That put work on Duvet, [Cricket](https://github.com/pybee/cricket) and [Bugjar](https://github.com/pybee/bugjar) on hold. However, Toga is now getting to the point where it can be used in practice, and we had a Google Summer of Code work on porting Cricket to Toga.

As for the best approach for pull requests - I'd suggest "large, thematically related" pull requests. For example, a pull request that cleans up all flake8 issues; or one that fixes a collection of Py3 porting issues. 

Essentially, the goal is to make each individual pull request easily reviewable. The number of changes isn't the critical factor - it's the number of *types* of change that need to be kept it the reviewer's head at any given time.

Does that make sense? Does it give you enough guidance to get underway?

(And, as a side note - thanks for taking an interest in Duvet! It means a lot to have someone share an interest in a project, especially when I haven't been able to give it the attention I want over the last couple of years)
